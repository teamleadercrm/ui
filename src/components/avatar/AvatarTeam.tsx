import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import cx from 'classnames';
import theme from './theme.css';
import Box from '../box';
import Icon from '../icon';
import { IconTeamMediumOutline, IconTeamSmallOutline } from '@teamleader/ui-icons';

class AvatarTeam extends PureComponent {
  render() {
    const { size } = this.props;

    return (
      <Box
        alignItems="center"
        backgroundColor="neutral"
        backgroundTint="darkest"
        className={cx(theme['avatar'], theme['avatar-team'])}
        data-teamleader-ui="avatar-team"
        display="flex"
        justifyContent="center"
      >
        <Icon color="neutral" tint="light">
          {size === 'tiny' || size === 'small' ? <IconTeamSmallOutline /> : <IconTeamMediumOutline />}
        </Icon>
      </Box>
    );
  }
}

AvatarTeam.propTypes = {
  /** The size of the avatar. */
  size: PropTypes.oneOf(['tiny', 'small', 'medium', 'large', 'hero']),
};

export default AvatarTeam;
