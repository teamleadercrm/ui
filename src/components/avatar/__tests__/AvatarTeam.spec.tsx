import React from 'react';
import { render } from '@testing-library/react';
import AvatarTeam from '../AvatarTeam';

describe('Component - AvatarTeam', () => {
  it('renders', async () => {
    const { asFragment } = render(<AvatarTeam />);
    expect(asFragment()).toMatchSnapshot();
  });
});
