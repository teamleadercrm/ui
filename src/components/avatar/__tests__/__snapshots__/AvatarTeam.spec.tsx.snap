// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component - AvatarTeam renders 1`] = `
<DocumentFragment>
  <div
    class="box align-items-center background-color-neutral-darkest display-flex justify-content-center avatar avatar-team"
    data-teamleader-ui="avatar-team"
  >
    <span
      class="box align-items-center display-flex neutral light"
      data-teamleader-ui="icon"
    >
      <svg
        fill="currentColor"
        height="24"
        preserveAspectRatio="xMidYMid meet"
        style="opacity: 0.84; vertical-align: middle;"
        viewBox="0 0 24 24"
        width="24"
      >
        <path
          d="M13 9.803A6 6 0 0122 15a1 1 0 11-2 0 4 4 0 00-7-2.646V9.803zM16 8a3 3 0 110-6 3 3 0 010 6zm0-2a1 1 0 100-2 1 1 0 000 2zM4 21a1 1 0 11-2 0 6 6 0 0112 0 1 1 0 11-2 0 4 4 0 00-8 0zm4-7a3 3 0 110-6 3 3 0 010 6zm0-2a1 1 0 100-2 1 1 0 000 2z"
          fill="currentColor"
        />
      </svg>
    </span>
  </div>
</DocumentFragment>
`;
